### PR TITLE
Move notification to pod controller

### PR DESCRIPTION
**Pull request by lpopo0856**
in this pr, 

- let `bitcoind` send notification to `pod controller` instead of `api server `
- make `pod controller` send the notification to `api server`

### CODE REVIEW

File: bitcoind/getrawtransaction.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	Vins          []Vin  `json:"vin"`
+	Vouts         []Vout `json:"vout"`
+	Hex           string `json:"hex"`
+	Blockhash     string `json:"blockhash"`
+	Confirmations int    `json:"confirmations"`
+	Time          int    `json:"time"`
+	Blocktime     int    `json:"blocktime"`
+}
+
+func (c *HttpBitcoind) GetRawTransaction(txId string) (*RawTransaction, error) {
```
Comment (nodestory) [inline bitcoind/getrawtransaction.go:45]: Could be done by the client from `NewBtcdRPCClient`.

File: bitcoind/gettransaction.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+type TransactionDetails struct {
+	Address  string  `json:"address"`
+	Category string  `json:"category"`
+	Amount   float64 `json:"amount"`
+	Label    string  `json:"label"`
+	Vout     int     `json:"vout"`
+}
+
+func (c *HttpBitcoind) GetTransaction(txId string) (*Transaction, error) {
```
Comment (nodestory) [inline bitcoind/gettransaction.go:39]: Could be done by the client from `NewBtcdRPCClient`.

File: notification.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	if err := json.NewEncoder(body).Encode(notifyFormat{AccountID: c.ownerDID, Contents: notifyContent, Data: notifyData}); err != nil {
+		logFields := map[string]interface{}{
+			"notifyData": notifyData,
+		}
+		replyWithError(context, err, "failed to encode tx", logFields)
+		return
+	}
+
+	// start to call notification api
+	notifyURL := viper.GetString("messaging.endpoint") + "/api/accounts/notification"
```
Comment (nodestory) [inline notification.go]: `messaging.endpoint` should be used only for the messaging module. I might configure this as a `notifictaion.tx_callback_url`.

File: bitcoind/bitcoind.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package bitcoind
```
Comment (nodestory) [inline bitcoind/bitcoind.go:6]: The license is missing.

**Review by nodestory — CHANGES_REQUESTED** [review]
Add the license for `bitcoind/bitcoind.go`.

For explicit Bitcoin RPC calls in the pod controller, I'd suggest using the client from btcd to make it simpler. I don't strongly request changes on these.

File: bitcoind/getrawtransaction.go (hunk 1 above)
Comment (lpopo0856) [reply bitcoind/getrawtransaction.go:45]: > Add the license for `bitcoind/bitcoind.go`.
> 
> For explicit Bitcoin RPC calls in the pod controller, I'd suggest using the client from btcd to make it simpler. I don't strongly request changes on these.

I view the btcd client, and there are some param missing in GetTransaction, so I have to use our own client to call the rpc with correct params.

**Comment by lpopo0856** [comment]
> Add the license for `bitcoind/bitcoind.go`.
> 
> For explicit Bitcoin RPC calls in the pod controller, I'd suggest using the client from btcd to make it simpler. I don't strongly request changes on these.

I view the btcd client, and there are some param missing in GetTransaction, so I have to use our own client to call the rpc with correct params.

File: config.yaml.sample [hunk 5]
```diff
@@ -27,3 +27,8 @@ bitcoind:
 bitcoind_ctl:
   endpoint: http://localhost:8888/bitcoind
   suspending_duration: 
+
+server_port: :8011
+
+notification:
+  tx_callback_url: https://autonomy-wallet.bitmark.com
```
Comment (nodestory) [inline config.yaml.sample]: If the name is `tx_callback_url`, the value should be `https://autonomy-wallet.bitmark.com/api/accounts/notification`.

Comment (nodestory) [reply config.yaml.sample]: And I just realized that the URL should accept all kinds of notification. Probably `notification_url` with enough succinctness.

**Review by nodestory — APPROVED** [review]

**Review by lemonlatte — APPROVED** [review]

**Pull request merged**